code/game/objects/items/weapons/chrono_eraser.dm

```diff
@@ -0,0 +1,267 @@
+
+/obj/item/weapon/gun/energy/chrono_gun/New(var/obj/item/weapon/chrono_eraser/T)
+	. = ..()
+	if(istype(T))
+		TED = T
+	else //admin must have spawned it
+		TED = new(src.loc)
+		qdel(src)
+
+/obj/item/weapon/gun/energy/chrono_gun/dropped()
+	qdel(src)
+
+/obj/item/weapon/gun/energy/chrono_gun/update_icon()
+	return
+
+/obj/item/weapon/gun/energy/chrono_gun/process_fire()
+	if(field)
+		field_disconnect(field)
+	..()
+
+/obj/item/weapon/gun/energy/chrono_gun/Destroy()
+	if(TED)
+		TED.PA = null
+		TED = null
+	if(field)
+		field_disconnect(field)
+	..()
+
+/obj/item/weapon/gun/energy/chrono_gun/proc/field_connect(var/obj/effect/chrono_field/F)
+	var/mob/living/user = src.loc
+	if(F.gun)
+		if(isliving(user) && F.captured)
+			user << "<span class='alert'><b>FAIL: <i>[F.captured]</i> already has an existing connection.</b></span>"
+		src.field_disconnect(F)
+	else
+		startpos = get_turf(src)
+		field = F
+		F.gun = src
+		if(isliving(user) && F.captured)
+			user << "<span class='notice'>Connection established with target: <b>[F.captured]</b></span>"
+
+
+/obj/item/weapon/gun/energy/chrono_gun/proc/field_disconnect(var/obj/effect/chrono_field/F)
+	if(F && field == F)
+		var/mob/living/user = src.loc
+		if(F.gun == src)
+			F.gun = null
+		if(isliving(user) && F.captured)
+			user << "<span class='alert'>Disconnected from target: <b>[F.captured]</b></span>"
+	field = null
+	startpos = null
+
+/obj/item/weapon/gun/energy/chrono_gun/proc/field_check(var/obj/effect/chrono_field/F)
+	if(F)
+		if(field == F)
+			var/turf/currentpos = get_turf(src)
+			var/mob/living/user = src.loc
+			if((currentpos == startpos) && (field in view(CHRONO_BEAM_RANGE, currentpos)) && !user.lying && (user.stat == CONSCIOUS))
+				return 1
+		field_disconnect(F)
+		return 0
+
+/obj/item/weapon/gun/energy/chrono_gun/proc/pass_mind(var/datum/mind/M)
+	if(TED)
+		TED.pass_mind(M)
+
+
+/obj/item/projectile/energy/chrono_beam
+	name = "eradication beam"
+	icon_state = "chronobolt"
+	range = CHRONO_BEAM_RANGE
+	color = null
+	nodamage = 1
+	var/obj/item/weapon/gun/energy/chrono_gun/gun = null
+
+/obj/item/projectile/energy/chrono_beam/fire()
+	gun = firer.get_active_hand()
+	if(istype(gun))
+		return ..()
+	else
+		return 0
+
+/obj/item/projectile/energy/chrono_beam/on_hit(var/atom/target)
+	if(target && gun && isliving(target))
+		var/obj/effect/chrono_field/F = new(target.loc, target, gun)
+		gun.field_connect(F)
+
+
+/obj/item/ammo_casing/energy/chrono_beam
+	name = "eradication beam"
+	projectile_type = /obj/item/projectile/energy/chrono_beam
+	icon_state = "chronobolt"
+	e_cost = 0
+
+
+/obj/effect/chrono_field
+	name = "eradication field"
+	desc = "An aura of time-bluespace energy."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "chronofield"
+	density = 1
+	anchored = 1
+	unacidable = 1
+	blend_mode = BLEND_MULTIPLY
+	var/mob/living/captured = null
+	var/obj/item/weapon/gun/energy/chrono_gun/gun = null
+	var/tickstokill = 30
+	var/icon/mob_snapshot = null
```

Comment:
Let me tell you I tried my damned hardest to avoid it, but in the end it didn't work out. At least the only lag you'll get if any is just for a moment (when the shot hits).